Docs 6939 tokenization january2020 (#44)
mulesoft/docs-anypoint-security: modules/ROOT/pages/create-tokenization-service.adoc

= Create a Tokenization Service
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]

The tokenization service enables you to substitute a sensitive data element with a non-sensitive equivalent.

== Prerequisites

To configure and use the tokenization service, you must:

> * Have Runtime Fabric, Anypoint Security Edge Policies, and Security Tokenization entitlements for your Anypoint Platform account. If you do not see the *Tokenization Service* tab in Runtime Manager, contact your customer success manager to enable the tokenization service for your account. 

Reviewer: One sentence max. per list item or step:
```suggestion
* Have Runtime Fabric, Anypoint Security Edge Policies, and Security Tokenization entitlements for your Anypoint Platform account
If you do not see the *Tokenization Service* tab in Runtime Manager, contact your customer success manager to enable the tokenization service for your account. 
```

> * Have the correct permissions to manage tokenization. See xref:grant-tokenization-permissions.adoc[Granting Tokenization Permissions]

Reviewer: ```suggestion
* Have the correct permissions to manage tokenization
See xref:grant-tokenization-permissions.adoc[Granting Tokenization Permissions].
```

> * Have Runtime Fabric 1.1.153 or later with inbound traffic configured. See xref:runtime-fabric::index.adoc[the Runtime Fabric documentation].

Reviewer: ```suggestion
* Have Runtime Fabric 1.1.153 or later with inbound traffic configured
See xref:runtime-fabric::index.adoc[the Runtime Fabric documentation].
```

> * Have a secret group to store the tokenization table encryption keys that are created by the tokenization service. See xref:asm-secret-group-concept.adoc[Secrets Manager documentation].

Reviewer: ```suggestion
* Have a secret group to store the tokenization table encryption keys that are created by the tokenization service
See xref:asm-secret-group-concept.adoc[Secrets Manager documentation].
```

> * Have a tokenization format, which describes the format and how data is tokenized. See xref:tokenization-formats.adoc[Tokenization Formats].

Reviewer: ```suggestion
* Have a tokenization format, which describes the format and how data is tokenized
See xref:tokenization-formats.adoc[Tokenization Formats].
```

> == What to Know About Secret Groups and Tokenization

Reviewer: Can we imply something more specific than "what to know about"? Are these requirements? restrictions? best practices?

Each tokenization service you create must have a corresponding, dedicated secret group in Secrets Manager. Do not use this secret group with any other tokenization service. Do not store any secrets outside of this tokenization service in another secret group.

This is the recommended approach for several reasons:

> . Use a dedicated secret group for the tokenization service to help prevent accidentally deleting other secret groups when you delete a secret group.

Reviewer: Unless there is a hierarchy here, use bullets instead of numbers.

Reviewer: I was looking for a reason right away, because of the intro, rather than an instruction, so how about this instead?
```suggestion
. Using a dedicated secret group for the tokenization service helps prevent accidentally deleting other secret groups when you delete a secret group.
```

. You can create and change as many formats as you need for the tokenization service, and the secret group will not be exhausted.

> . The unique association between the tokenization service and its dedicated secret group prevents vaultless data protection (VDP) type secrets from running out in a secret group. In the worst case, there will be 10 secrets created of the VDP Context type&#8212;

Reviewer: Can we omit "type" and still retain meaning? Keeping it is a little awkward, because it really should be hyphenated with the name before it: VDP-type secrets.
Also, suggest rewording "running out," which might cause translation issues. Is there a limit that this association prevents exceeding?
And why is 10 secrets a "worst case"?
```suggestion
* The unique association between the tokenization service and its dedicated secret group prevents exceeding the allowed number of vaultless data protection (VDP) secrets in a secret group. 
In the worst case, there will be 10 secrets created of the VDP Context type&#8212: one for each VDP domain.
```

> one for each VDP domain.

Reviewer: ```suggestion
```

Use Secrets Manager xref:asm-secret-group-creation-task.adoc[to create and manage secret groups]. 

> Tokenization table keys appear in in the *Shared Secret* tab in the secret group that is associated with the tokenization service. Normally, you can view and edit shared secrets in Secrets Manager, however, the tokenization table keys are a view-only, non-editable part of the secret group. 

Reviewer: ```suggestion
Tokenization table keys appear in the *Shared Secret* tab in the secret group that is associated with the tokenization service. Normally, you can view and edit shared secrets in Secrets Manager; however, the tokenization table keys are a view-only, noneditable part of the secret group. 
```

Reviewer: End the previous text with a colon if the end of that text refers directly to the image.

image::tokenization-example-shared-secret.png[]

Click *View* to see the name, type, and expiration of the shared secret.

== Create the Tokenization Service

> 

Reviewer: Is there any transition text that you would like to provide here to provide context for the steps beyond that provided by the title? Perhaps something like "After you understand the best practices that apply to secret groups and tokenization, you can create your tokenization service in Anypoint Runtime Manager:"? Is there anything to add about why, or the advantages to the user, that have not yet been mentioned?

> . Go to the *Runtime Manager­ > Tokenization Service* page to create the service.

Reviewer: ```suggestion
. Go to the *Runtime Manager­ > Tokenization Service* page.
```
No need to repeat the goal of the task in the first step.

. Click *Create Tokenization Service*. +

> The prerequisites you completed provide the information you need for the following information: 

Reviewer: ```suggestion
The prerequisites you completed provide the information you need to make the following selections: 
```
HOWEVER...I wonder if these are really substeps? After clicking *Create Tokenization Service*, is selecting the fabric instance not the next step?
If you promote the substeps, be sure to replace the colon at the end of the sentence here with a period.

>   .. In the *Runtime fabric* drop-down list, select the Runtime Fabric to which to deploy the tokenization service. 

Reviewer: Is the UI label for the list in mixed case: Runtime fabric? Just verifying.

Reviewer: Select the Runtime Fabric or Select the instance of Runtime Fabric? Runtime Fabric instance?

Reviewer: GLOBALS:

Do we really have to identify menus (which is what these are, I think, rather than "list"s) as "drop-down" or is that obvious to our audience?
Also, this PR alternately formats the names of menus in bold and in plain text. I prefer bold only for things the user actually clicks or selects, but the department standard has been to boldface everything, and I am getting pushback. For the sake of consistency, I am going to mark latter instances as bold as well.

>   .. Select the tokenization format from the *Formats* drop-down list. You can assign one, more, or all, formats to one tokenization service.

Reviewer: ```suggestion
  .. Select the tokenization format from the *Formats* menu. 
  You can assign one, more, or all formats to one tokenization service.
```
Alternative to consider: "You can assign one or more formats or all formats to one tokenization service."

>   .. Select the secret group from the *Secret Group* drop-down list.

Reviewer: ```suggestion
  .. Select the secret group from the *Secret Group* menu.
```

>   .. Select the number of tokenization service replicas to run. The tokenization service runs on worker nodes in Runtime Fabric.

Reviewer: ```suggestion
  .. Select the number of tokenization service replicas to run. 
  The tokenization service runs on worker nodes in Runtime Fabric.
```

>   .. Select the log level for the tokenization service, or keep the default, *ERROR*. 

Reviewer: ```suggestion
  .. Select the log level for the tokenization service or keep the default (*ERROR*). 
```

. Click *Build and Deploy* to create the tokenization table. +

> A mapping table that contains a large table of randomizations, which are used at the core of tokenizing and detokenizing, pre-builds. This pre-build is not a one-to-one table of mappings&#8212;it is used during internal steps to swap in and swap out randomizations in place of the actual data. 

Reviewer: ```suggestion
A mapping table builds, containing a large number of randomizations that are used at the core of tokenizing and detokenizing. This prebuilt table is not a table of one-to-one mappings; it is used during internal steps to swap in and swap out randomizations in place of the actual data. 
```
Please review this rewording carefully to ensure that your intended technical meaning has not been altered by the suggestions. 
Also, do you suppose that an illustration might help here?

> The tokenization mapping table build is a one-time action and takes some time to complete, depending on the size of the tokenization format. For example, a table that contains only an SSN format with a size that is less than 200 MB might build in two minutes; but a table that uses a larger format, such as “lax alphanumeric” might take up to 20 minutes to build.

Reviewer: ```suggestion
The tokenization mapping table build is a one-time action and takes some time to complete, depending on the size of the tokenization format. For example, a table that contains only an SSN format with a size that is less than 200 MB might build in 2 minutes, but a table that uses a larger format, such as “lax alphanumeric,” might take up to 20 minutes to build.
```
I'm not sure you need "lax alphanumeric" in quotation marks. It makes sense to me without them, as a relaxed alphanumeric scheme in general. But this might mean something different in this tech context?
The reason we change "two" to "2" here is that we don't want to mix spelling numbers and values with using digits together in one sentence.

> If many, or all, of the formats are selected, it takes much longer to build the table, as a table with all formats is approximately 2 GB in size.

Reviewer: ```suggestion
If many or all of the formats are selected, it takes much longer to build the table, because a table with all formats is approximately 2 GB in size.
```
